src/elements/content-explorer/ContentExplorer.js

```diff
@@ -758,36 +745,40 @@ class ContentExplorer extends Component<Props, State> {
      * @param {Function} [callback] - callback function that should be called after setState occurs
      * @return {void}
      */
-    updateCollection(collection: Collection, selected: ?BoxItem, callback: Function = noop): Object {
+    async updateCollection(collection: Collection, selectedItem: ?BoxItem, callback: Function = noop): Object {
+        const { features } = this.props;
+        const { items } = collection;
         const newCollection: Collection = { ...collection };
-        let newSelected: ?BoxItem;
-        const targetID = selected ? selected.id : null;
+        const selectedId = selectedItem ? selectedItem.id : null;
+        let newSelectedItem: ?BoxItem;
 
-        if (collection.items) {
-            newCollection.items = collection.items.map(obj => {
-                const item = { ...obj, selected: obj.id === targetID };
+        const isGridViewEnabled = isFeatureEnabled(features, 'contentExplorer.gridView.enabled');
 
-                // If the previously selected item is found in the folder, keep it as selected item
-                if (item.id === targetID) {
-                    newSelected = item;
+        if (items) {
+            let itemThumbnails;
+            if (isGridViewEnabled) {
+                const fileAPI = this.api.getFileAPI();
+                itemThumbnails = await Promise.all(items.map(item => fileAPI.getThumbnailUrl(item)));
```

Comment:
I wonder if we can change `getThumbnailUrl` to be a static method. We would need to pass in the token directly. I'm trying to get away from L760 because that always destroys the `APIFactory` and we call `updateCollection` a lot